backend/src/main/kotlin/org/loculus/backend/utils/EarliestReleaseDateFinder.kt

```diff
@@ -0,0 +1,74 @@
+package org.loculus.backend.utils
+
+import kotlinx.datetime.LocalDate
+import kotlinx.datetime.LocalDateTime
+import kotlinx.datetime.LocalTime
+import mu.KotlinLogging
+import org.loculus.backend.service.submission.RawProcessedData
+
+private val log = KotlinLogging.logger { }
+
+/**
+ * Calculate the earliest release date for rows of sequence entries given to it one by one.
+ * Assumes that rows are sorted: all accession entries are given in a block, and with ascending versions.
+ *
+ * The earliest release date of a sequence is the earliest date of:
+ * - the internal release date
+ * - any date from a given list of fields
+ * - the earliest release date from the previous version (if it exists)
+ */
+class EarliestReleaseDateFinder(private val fields: List<String>) {
```

Comment:
Does this need to be so complicated? 

Earliest release date of Loculus is always the release date of version 1. We can get this at db level in the view.

Then calculating the minimum of that and whatever else comes in from processed data is trivial as it can be done at the single sequence level.

Reply:
The specifics of calculating the earliest release date can be changed (like you said) quite easily, and it's all in this class, and there are tests for the class. Not sure what you find complicated about it.